eo-maven-plugin/src/main/java/org/eolang/maven/Cached.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2022 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import org.cactoos.text.TextOf;
+
+/**
+ * Cached file in EO compilation process. The file is assumed to
+ * have text content in any format.
+ * @since 1.0
+ */
+public class Cached {
```

Comment:
@mximp what's they reason for this class? It seems like a collection of functions around a `path()` method. Maybe it will be more readable if we just inline this functionality into `ParseMojo`.

Reply:
@yegor256 
- first, this would make `ParseMojo` to have quite a few static methods which are not directly related to mojo logic. Caching in this case seems to be quite self-contained logic which deserves dedicated object. Isn't it a good OO style?
- also we will definitely want to reuse this functionality for other caching cases (optimized files for example).

Reply:
@mximp it's a good style to have dedicated objects for smaller tasks, for sure. But in this case, there is almost no logic in this object, it's a bag of methods. If you want to have a special object, I would let it deal with the entire caching functionality:

```
String program-name = "org/eolang/string.eo";
Footprint f = new Footprint(main-dir, cache-dir, git-hash);
f.save(program-name, content);
String content = f.load(program-name);
```

The class `Footprint` will _behind the scene_ make the decision about where to write and where to read, based on the git hash and program name.

Reply:
@yegor256 How is this conceptually different from suggested `Cached`?
As I see `Footprint`:
- encapsulates cache root, hash/ver and object name (I din't get the meaning of `main-dir` parameter)
- allows to save content
- allows to retrieve content
So as `Cache` does.  It's just in `Footprint` you moved object name  into method arg which allows to use single `Footprint` instance for many objects. But functionally the logic behind save/load will be the same: it is still pure operations on paths and files.
What logic do you mean by _'entire caching functionality'_ ?
Also `Cache` does make decision on where to read/write.

Reply:
@yegor256 Just to compare, the above code would look like the following with `Cached`:
```java
String program-name = "org.eolang.int.xmir";
Cached cached = new Cached(git-hash, program-name, cache-dir);
cached.save();
String content = cached.content();
```

Reply:
@mximp not really. Your current `save()` method only saves to cache, while saving to the main location is done outside of `Cached`. I suggest encapsulating the entire logic into the `Footprint` class.